lib/backend/backend.go

```diff
@@ -244,6 +245,14 @@ func NextPaginationKey(r types.Resource) string {
 	return string(nextKey([]byte(r.GetName())))
 }
 
+// MaskKeyName masks the given key name.
+// e.g "123456789" -> "******789"
+func MaskKeyName(str string) string {
+	hiddenBefore := int(math.Floor(0.75 * float64(len(str))))
+	asterisks := bytes.Repeat([]byte("*"), hiddenBefore)
+	return string(append(asterisks, str[hiddenBefore:]...))
```

Comment:
Depending on how frequently this is called, you may want to accept a `[]byte` instead of a string, and mask the data out by mutating the slice.

Strings allocate, and if this is called frequently as we stream logs we're generating a lot of tiny strings for the GC to clean up.

Reply:
From looking at the call site, the source data already seems to be a string. Copying the `string` into a `[]byte` doesn't really save you anything, especially if you have to copy it back to a `string` afterwards.

Reply:
Right, but you don't need to copy it back to a `string` afterwords if all you're going to do is write it to the logger.  `%s` will print a `[]byte` just fine.

Really not a big deal unless we expect to be masking in hundreds of thousands of log lines - in which case all those tiny allocations add up.

Reply:
If I understand stack allocation correctly in go, this shouldn't be a concern. The `str` parameter will be copied and allocated to the stack, not the heap, so as soon as `MaskKeyName` returns, the copied `str` value and other allocated values within the function will be popped off the stack without any work for the GC.